tests/PolyType.SourceGenerator.UnitTests/CompilationTests.cs

```diff
@@ -266,6 +266,25 @@ public partial record DerivedClassWithShadowingMember : BaseClassWithShadowingMe
         Assert.Empty(result.Diagnostics);
     }
 
+    [Fact]
+    public static void IsRequiredProperty()
+    {
+        Compilation compilation = CompilationHelpers.CreateCompilation("""
+            using PolyType;
+            #nullable enable
+
+            [GenerateShape]
+            public partial class cFoo
+            {
+                [PropertyShape(IsRequired = true)]
```

Comment:
Do we have a test with a `required` property set to `IsRequired = false`?

Reply:
Yes:

https://github.com/eiriktsarpalis/PolyType/blob/b7a7008bb1eed90dba5cc1d8c78e08f0d08d84fb/src/PolyType.TestCases/TestTypes.cs?plain=1#L2289C1-L2295C1